### DISCUSSION TITLE
ideajoin doesnt work

### DISCUSSION THREAD

**Discussion by comod** [JetBrains/ideavim#432] [Q&A] [answered]
ive added
`set ideajoin` to ~/.ideavimrc
but on press `J` the lines will be  joined together, but not smart like if (a && b)

**Comment by comod** [comment]
I also added these
```
clipboard ideaput
set ideamarks
```
But it does not work

**Reply by AlexPl292** [reply]
Hi, in what language do you try to do `J` command? Could you please also try to use ctrl-shift-j, which is IJ smart join direct shortcut. It may happen that joining ifs is not supported by this language yet.

![image](https://user-images.githubusercontent.com/4203721/148941572-be833538-d7de-4963-ad5e-03be98f0e9d7.png)

**Reply by comod** [reply]
yeah i tested all this. i tried .ts & .php
eg:
```
if (true) {
    if (false) {
        //hj
    }
}
```
just becomes

```
if (true) {if (false) {
        //hj
    }
}
```

plain editor-action do the same

**Reply by AlexPl292** [reply]
Well, I guess, that this feature is just not implemented for TypeScript. It was implemented some time ago for JavaScript https://youtrack.jetbrains.com/issue/WEB-48537
And I've created a ticket for TypeScript.

https://youtrack.jetbrains.com/issue/WEB-54358

**Reply by comod** [reply]
But php also not work?!

**Reply by AlexPl292** [reply]
Yep, and this one for php https://youtrack.jetbrains.com/issue/WEB-54360

**Reply by comod** [reply]
Lol. K thx

**Comment by comod** [comment, marked as answer]
TL;DR:
https://youtrack.jetbrains.com/issue/WI-64909
https://youtrack.jetbrains.com/issue/WEB-54358
https://youtrack.jetbrains.com/issue/WEB-48537
